src/bcd/bcd/DenoisingUnit.cpp

```diff
@@ -91,124 +91,46 @@ void DenoisingUnit::denoisePatchAndSimilarPatches(
 {
     m_mainPatchCenter = i_rMainPatchCenter;
     {
-        float skippingProbability =
-                m_rDenoiser.getParameters().m_markedPixelsSkippingProbability;
-        if (skippingProbability != 0)
-            if (m_pIsCenterOfAlreadyDenoisedPatchImage->getValue(m_mainPatchCenter,
-                                                                 0))
-                if (skippingProbability == 1 ||
-                    rand() < static_cast<int>(skippingProbability * RAND_MAX))
+        float skippingProbability = m_rDenoiser.getParameters().m_markedPixelsSkippingProbability;
```

Comment:
I'm not sure we want to do any changes in the BCD code. It's usually better to leave foreign code completely unmodified: it makes it simpler to diff between our version and subsequent ones, and to update the code when new versions are released.

Reply:
I am not sure if BCD is maintained. 
I already did changes to remove dependencies, fix some small bugs and I plan to do some improvements soon.